client/context/GlobalContext.tsx

```diff
@@ -149,3 +175,50 @@ export default {
     GlobalContextProvider,
     useGlobalContext
 }
+
+/* User Theme */
+
+// Get system theme from CSS media queries
+function systemTheme(): ThemeOption {
+    let defaultTheme = "light" as ThemeOption;
+
+    if (typeof window != "undefined" && typeof window.matchMedia != "undefined") {
+        if (window.matchMedia("(prefers-color-scheme)").media == "not all") {
+            return defaultTheme;
+        }
+
+        const isDark = !window.matchMedia("(prefers-color-scheme: light)").matches;
+        return (isDark ? "dark" : "light")  as ThemeOption;
+    }
+
+    return defaultTheme;
+}
+
+function applyUserThemeToBody(userTheme: ThemeOption) {
+    if (typeof window != "undefined") {
+        document.body.setAttribute(
+            "color-scheme",
+            userTheme == "system" ? systemTheme() : userTheme
+        );
+    }
+}
+
+// Save page theme to localStorage
+function localSaveUserTheme(mediaTheme: ThemeOption) {
+    if (typeof window != "undefined") {
+        localStorage.setItem("vudl-theme", mediaTheme);
+    }
+}
+
+// Get page theme from localStorage
+function localLoadUserTheme(): ThemeOption {
+    if (typeof window != "undefined") {
+        let mediaTheme = (localStorage.getItem("vudl-theme") ?? "system") as ThemeOption;
+
+        applyUserThemeToBody(mediaTheme);
+
+        return mediaTheme;
+    }
+
+    return "system" as ThemeOption;
```

Comment:
Can you return `ThemeOption.system` instead of `"system" as ThemeOption`? If that works, it seems easier to understand. (And we should use that pattern for all the theme references).